tests/unit_tests/parsers/document/test_document.py

```diff
@@ -3,7 +3,201 @@
+            "title": "Principles and Guidelines for Sustainable Urban Development",
+            "variant_name": "Translated",
+        },
+        {
+            "metadata": {
+                "type": "Climate Action Plans",
+            },
+            "source_url": "https://www.climateprojects.org/docs/zh-sustainable-urban-development.pdf",
+            "title": "Principles and Guidelines for Sustainable Urban Development",
+            "variant_name": "Translated",
+        },
+        {
+            "metadata": {
+                "type": "Climate Action Plans",
+            },
+            "source_url": "https://www.climateprojects.org/document/climate-resilience-framework",
+            "title": "Framework for Climate Resilience and Adaptation",
+            "variant_name": "Original Translation",
+        },
+    ]
+
+
+@pytest.fixture
+def parsed_document_data_with_no_translated_titles():
+    return [
+        {
+            "metadata": {
+                "type": "Climate change mitigation project",
+            },
+            "source_url": "https://www.climateprojects.org/solar-power-project",
+            "title": "Community-based solar power project",
+            "variant_name": "Original Translation",
+        },
+        {
+            "metadata": {
+                "type": "Renewable energy initiative",
+            },
+            "source_url": "https://www.climateprojects.org/reforestation-program",
+            "title": "Afforestation and reforestation program",
+            "variant_name": "Original Translation",
+        },
+        {
+            "metadata": {
+                "type": "Sustainable agriculture program",
+            },
+            "source_url": "https://www.climateprojects.org/organic-farming-project",
+            "title": "Organic farming and soil conservation project",
+            "variant_name": "Original Translation",
+        },
+        {
+            "metadata": {
+                "type": "Carbon offset project",
+            },
+            "source_url": "https://www.climateprojects.org/mangrove-conservation-project",
+            "title": "Mangrove restoration and conservation project",
+            "variant_name": "Original Translation",
+        },
+    ]
+
+
+def test_returns_expected_value(
+    test_document_df_without_translated_docs,
+    parsed_document_data_with_no_translated_titles,
+):
+    document_data = document(test_document_df_without_translated_docs, debug=True)
+    assert document_data is not None
+    assert document_data == parsed_document_data_with_no_translated_titles
+
+
+def test_returns_expected_value_for_translated_titles(
+    test_document_df_with_translated_docs, parsed_document_data_with_translated_titles
+):
+    document_data = document(test_document_df_with_translated_docs, debug=True)
+    assert document_data is not None
+    assert document_data == parsed_document_data_with_translated_titles
```

Comment:
Perhaps we can use the proper pandas testing modules here -- [pandas.testing.assert_frames_equal](https://pandas.pydata.org/docs/reference/api/pandas.testing.assert_frame_equal.html) for example